lib/annealing/simulator.rb

```diff
@@ -3,62 +3,50 @@
 module Annealing
   # It runs simulated annealing
   class Simulator
-    include Configuration::Configurator
+    attr_reader :configuration
 
-    def initialize(**config)
-      init_configuration(config)
+    def initialize(config_hash = {})
+      @configuration = Annealing.configuration.merge(config_hash)
```

Comment:
I think it will be better the keep the variable name config since its only in the initializer
```suggestion
    def initialize(config = {})
      @configuration = Annealing.configuration.merge(config)
```

Reply:
My only worry with that is it could lead to confusion about whether it expects another `Configuration` instance versus a hash of configuration values. This also matches the argument to the `run` method later on. The `Metal` class initializer accepts a config argument too, but in that case it is actually a `Configuration` instance.